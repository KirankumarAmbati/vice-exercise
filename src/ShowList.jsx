import React from 'react';
import PropTypes from 'prop-types';
import styles from './showList.less';

const ShowList = ({ shows, handler }) => (
  <div className="showList">
    <div onClick={() => handler(-2)} className="item skinny" />
    <div onClick={() => handler(-1)} className="item" />
    <div className="item focused" />
    <div onClick={() => handler(1)} className="item" />
    <div onClick={() => handler(2)} className="item skinny" />
  </div>
);

export default ShowList;
