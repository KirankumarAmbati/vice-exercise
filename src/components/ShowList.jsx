import React from 'react';
import PropTypes from 'prop-types';

const base = 'show-list';
const ShowList = ({ handler, activeIndex, rightBound }) => {
  const items = Array.from({ length: 5 }, (el, i) => {
    const shift = activeIndex + -2 + i;
    const hide = shift < 0 || shift >= rightBound;

    return (
      <button
        tabIndex="0"
        key={i}
        onClick={() => handler(shift)}
        className={`item${hide ? ' hide' : ''}`}
      />
    );
  });

  return (
    <div className={base}>
      <div className={`${base}--items`}>{items}</div>
      <div className={`${base}--page`}>{activeIndex + 1}</div>
    </div>
  );
};

export default ShowList;

ShowList.propTypes = {
  handler: PropTypes.func.isRequired,
  activeIndex: PropTypes.number.isRequired,
  rightBound: PropTypes.number.isRequired,
};
