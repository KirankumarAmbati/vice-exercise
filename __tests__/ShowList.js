import React from 'react';
import renderer from 'react-test-renderer';
import ShowList from '../src/components/ShowList';

const handler = () => { };
test('ShowList component baseline', () => {
  const component = renderer.create(<ShowList handler={handler} activeIndex={3} rightBound={10} />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('ShowList component edge case', () => {
  const component = renderer.create(<ShowList handler={handler} activeIndex={9} rightBound={10} />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
