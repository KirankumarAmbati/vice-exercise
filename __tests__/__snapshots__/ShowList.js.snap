// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowList component baseline 1`] = `
<div
  className="show-list"
>
  <div
    className="show-list--items"
  >
    <button
      className="item"
      onClick={[Function]}
      tabIndex="0"
    />
    <button
      className="item"
      onClick={[Function]}
      tabIndex="0"
    />
    <button
      className="item"
      onClick={[Function]}
      tabIndex="0"
    />
    <button
      className="item"
      onClick={[Function]}
      tabIndex="0"
    />
    <button
      className="item"
      onClick={[Function]}
      tabIndex="0"
    />
  </div>
  <div
    className="show-list--page"
  >
    4
  </div>
</div>
`;

exports[`ShowList component edge case 1`] = `
<div
  className="show-list"
>
  <div
    className="show-list--items"
  >
    <button
      className="item"
      onClick={[Function]}
      tabIndex="0"
    />
    <button
      className="item"
      onClick={[Function]}
      tabIndex="0"
    />
    <button
      className="item"
      onClick={[Function]}
      tabIndex="0"
    />
    <button
      className="item hide"
      onClick={[Function]}
      tabIndex="0"
    />
    <button
      className="item hide"
      onClick={[Function]}
      tabIndex="0"
    />
  </div>
  <div
    className="show-list--page"
  >
    10
  </div>
</div>
`;
